### PR TITLE
Deprecated code and update dependencies

### PR DESCRIPTION
**Pull request by raych1**
- Removed dependencies of swagger-validation-common and azure-js-dev-tools
- Deprecated code for supporting track1 SDK generation
- Deleted invalid tests
- Updated dependencies
- Added pipelines.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: tools/spec-gen-sdk/src/types/sdks.ts
Comment (maririos) [inline tools/spec-gen-sdk/src/types/sdks.ts:15]: could you remind me the difference between `azure-sdk-for-net` and `azure-sdk-for-net-track2` ?

File: tools/spec-gen-sdk/src/utils/readme.ts
Comment (maririos) [inline tools/spec-gen-sdk/src/utils/readme.ts:57]: is this something that could be used by other jobs in the specs repo that would make more sense to have in there than in here?

**Review by maririos — APPROVED** [review]

File: tools/spec-gen-sdk/src/types/sdks.ts
Comment (weshaggard) [reply tools/spec-gen-sdk/src/types/sdks.ts:15]: I would love to remove this "-track2" as part of this work if possible. IIRC the older was for the old code generator.

Comment (maririos) [reply tools/spec-gen-sdk/src/types/sdks.ts:15]: same. This causes so much confusion that if we can use this effort to clean it up, I am all up for it!

**Review by mikeharder — CHANGES_REQUESTED** [review]
Please wait for my review

**Comment by weshaggard** [comment]
@raych1 are we at a point where we can have a buildable pipeline? If so lets try to get it added.

File: tools/spec-gen-sdk/src/utils/readme.ts
Comment (raych1) [reply tools/spec-gen-sdk/src/utils/readme.ts:57]: It might be used in other place where requires to access the SDK configuration in `readme.md`. But we don't have an individual tool staying in spec repo and is published as a library in order to be consumed.

File: tools/spec-gen-sdk/src/types/sdks.ts
Comment (raych1) [reply tools/spec-gen-sdk/src/types/sdks.ts:15]: Thank you for the suggestion. I like it. I will confirm with Arthur if the name can be updated. Otherwise, I prefer to keep the name consistent with what they use.

**Comment by raych1** [comment]
> @raych1 are we at a point where we can have a buildable pipeline? If so lets try to get it added.

That's the next PR I am planning to add.

File: tools/spec-gen-sdk/src/utils/readme.ts
Comment (maririos) [reply tools/spec-gen-sdk/src/utils/readme.ts:57]: Does the flow before this involves cloning the specs repo? What if there is a common script on the spec repo that different tools can call? no need to publish anything

File: tools/spec-gen-sdk/package.json
Comment (mikeharder) [inline tools/spec-gen-sdk/package.json:2]: Will this package be published to npmjs?  If so, I believe our convention is to use prefix `@azure-tools`.  Example:

https://github.com/Azure/azure-sdk-tools/blob/main/tools/js-sdk-release-tools/package.json

File: tools/spec-gen-sdk/package.json
Comment (mikeharder) [inline tools/spec-gen-sdk/package.json:32]: This is still a large number of dependencies.  Do you believe they are all needed?

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: tools/spec-gen-sdk/package.json
Comment (raych1) [reply tools/spec-gen-sdk/package.json:33]: Yes, I walked through the list, and they are being used.

File: tools/spec-gen-sdk/package.json
Comment (raych1) [reply tools/spec-gen-sdk/package.json]: Yes, I updated it along with the pipeline definitions.

File: tools/spec-gen-sdk/src/utils/readme.ts
Comment (raych1) [reply tools/spec-gen-sdk/src/utils/readme.ts:57]: The new design won't have dependency on the spec repo. It makes sense to keep this function in the same code base or reference it from another npm library.

**Comment by raych1** [comment]
@weshaggard 
could you review the pipeline changes?

File: tools/spec-gen-sdk/ci.yml
Comment (weshaggard) [inline tools/spec-gen-sdk/ci.yml:35]: These variables don't seem to be used so I'd suggest removing them.

File: tools/spec-gen-sdk/ci.yml
Comment (weshaggard) [inline tools/spec-gen-sdk/ci.yml:18]: Is there a reason you want to have this publish parameter? It will not actually control the real publishing at https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/stages/archetype-sdk-publish-js.yml#L24. 

I would suggest removing it.

File: tools/spec-gen-sdk/CHANGELOG.md
Comment (weshaggard) [inline tools/spec-gen-sdk/CHANGELOG.md:1]: Please also add an entry in the CODEOWNERS file

File: tools/spec-gen-sdk/ci.yml
Comment (raych1) [reply tools/spec-gen-sdk/ci.yml]: No plan at least currently. Let me add them inline.

**Review by weshaggard — APPROVED** [review]

**Comment by raych1** [comment]
@mikeharder do you have any further feedback on this PR?

File: tools/spec-gen-sdk/package.json
Comment (mikeharder) [reply tools/spec-gen-sdk/package.json:33]: It can happen in a later PR, but I think we should try to trim this down.  It seems unnecessarily large for what the tool needs to do.

**Review by mikeharder — APPROVED** [review]

File: tools/spec-gen-sdk/src/types/sdks.ts
Comment (raych1) [reply tools/spec-gen-sdk/src/types/sdks.ts:15]: I have initiated the discussion with Shanghai team on this name repurpose and will update this code if decided.

Comment (maririos) [reply tools/spec-gen-sdk/src/types/sdks.ts:15]: Thank you!

**Pull request merged**
